x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/analysis/VerifierTests.java

```diff
@@ -1079,83 +1080,155 @@ public void testMatchFilter() throws Exception {
     public void testQueryStringFunctionsNotAllowedAfterCommands() throws Exception {
         assumeTrue("skipping because QSTR is not enabled", EsqlCapabilities.Cap.QSTR_FUNCTION.isEnabled());
 
-        // Source commands
-        assertEquals("1:13: [QSTR] function cannot be used after SHOW", error("show info | where qstr(\"8.16.0\")"));
+        assertEquals("1:13: [QSTR] function cannot be used after SHOW", error("show info | where qstr(\"Anna\")"));
         assertEquals("1:17: [QSTR] function cannot be used after ROW", error("row a= \"Anna\" | where qstr(\"Anna\")"));
+        checkFullTextFunctionNotAllowedAfterCommands("QSTR", "qstr(\"Anna\")");
+    }
+
+    public void testMatchFunctionNotAllowedAfterCommands() throws Exception {
+        assumeTrue("skipping because MATCH is not enabled", EsqlCapabilities.Cap.MATCH_FUNCTION.isEnabled());
+
+        assertEquals("1:13: [MATCH] function cannot be used after SHOW", error("show info | where match(version, \"8.*\")"));
+        checkFullTextFunctionNotAllowedAfterCommands("MATCH", "match(first_name, \"Anna\")");
+    }
+
+    private void checkFullTextFunctionNotAllowedAfterCommands(String functionName, String functionInvocation) throws Exception {
+
+        // Source commands
+        String function = functionName.toLowerCase(Locale.ROOT);
 
         // Processing commands
         assertEquals(
-            "1:43: [QSTR] function cannot be used after DISSECT",
-            error("from test | dissect first_name \"%{foo}\" | where qstr(\"Connection\")")
+            "1:43: [" + functionName + "] function cannot be used after DISSECT",
+            error("from test | dissect first_name \"%{foo}\" | where " + functionInvocation)
         );
-        assertEquals("1:27: [QSTR] function cannot be used after DROP", error("from test | drop emp_no | where qstr(\"Anna\")"));
         assertEquals(
-            "1:71: [QSTR] function cannot be used after ENRICH",
-            error("from test | enrich languages on languages with lang = language_name | where qstr(\"Anna\")")
+            "1:27: [" + functionName + "] function cannot be used after DROP",
+            error("from test | drop emp_no | where " + functionInvocation)
         );
-        assertEquals("1:26: [QSTR] function cannot be used after EVAL", error("from test | eval z = 2 | where qstr(\"Anna\")"));
```

Comment:
Do we still have a test for `EVAL` for QSTR?
For `match` we would want to allow it to be used after an `EVAL` - but we would need to explicitly check that the field it receives is a field present in the mappings similar to what we did for the `match` operator:

https://github.com/elastic/elasticsearch/blob/ade235bf0f032926ab611a17d553160bc2f254a9/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java#L622